pkg/operator/staticresource/controller.go

```diff
@@ -0,0 +1,289 @@
+			return isManaged(e.Meta)
+		},
+		DeleteFunc: func(e event.DeleteEvent) bool {
+			return isManaged(e.Meta)
+		},
+	}
+
+	cache := mgr.GetCache()
+	allFiles := append(staticFiles, templateFiles...)
+	for _, file := range allFiles {
+		objBytes := v410_00_assets.MustAsset(file)
+		obj, _, err := defaultCodec.Decode(objBytes, nil, nil)
+		if err != nil {
+			return err
+		}
+
+		informer, err := cache.GetInformer(obj)
+		if err != nil {
+			return err
+		}
+
+		err = c.Watch(&source.Informer{Informer: informer}, &handler.EnqueueRequestForObject{}, p)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func isManaged(meta metav1.Object) bool {
+	// all managed resources are named pod-identity-webhook
+	if meta.GetName() == "pod-identity-webhook" && meta.GetNamespace() == operatorNamespace {
+		return true
+	}
+	// we also care about service-ca rotation
+	if meta.GetName() == "service-ca" && meta.GetNamespace() == "openshift-config-managed" {
+		return true
+	}
+	return false
+}
+
+type staticResourceReconciler struct {
+	clientset                              *kubernetes.Clientset
+	logger                                 log.FieldLogger
+	eventRecorder                          events.Recorder
+	deploymentGeneration                   int64
+	mutatingWebhookConfigurationGeneration int64
+	forceRollout                           bool
+}
+
+var _ reconcile.Reconciler = &staticResourceReconciler{}
+
+func (r *staticResourceReconciler) Reconcile(request reconcile.Request) (reconcile.Result, error) {
+	log.WithField("controller", controllerName).Debugf("reconciling after watch event %#v", request)
+	if request.Name == "service-ca" && request.Namespace == "openshift-config-managed" {
+		log.WithField("controller", controllerName).Infof("%s configmap change detected", request.NamespacedName)
+		r.forceRollout = true
+	}
+	r.ReconcileResources()
+	return reconcile.Result{}, nil
+}
+
+func (r *staticResourceReconciler) ReconcileResources() {
+	applyResults := resourceapply.ApplyDirectly(
+		(&resourceapply.ClientHolder{}).WithKubernetes(r.clientset),
+		r.eventRecorder,
+		v410_00_assets.Asset,
+		staticFiles...,
+	)
+	for _, result := range applyResults {
+		if result.Error != nil {
+			log.WithField("controller", controllerName).WithError(result.Error).Errorf("error reconciling %s", result.Type)
+			continue
+		}
+		if !result.Changed {
+			continue
+		}
+		log.WithField("controller", controllerName).Infof("%s reconciled successfully", result.Type)
+	}
+
+	// "v4.1.0/aws-pod-identity-webhook/deployment.yaml"
+	requestedDeployment := resourceread.ReadDeploymentV1OrDie(v410_00_assets.MustAsset("v4.1.0/aws-pod-identity-webhook/deployment.yaml"))
+	requestedDeployment.Spec.Template.Spec.Containers[0].Image = os.Getenv("AWS_POD_IDENTITY_WEBHOOK_IMAGE")
+	resultDeployment, modified, err := resourceapply.ApplyDeployment(r.clientset.AppsV1(), r.eventRecorder, requestedDeployment, r.deploymentGeneration, r.forceRollout)
+	r.forceRollout = false
```

Comment:
Could some strange things happen here with this forceRollout state being set on the reconciler if we were reconciling concurrently?

Reply:
I think I'm going to pull the watching of the `service-ca` and redeployment logic for this PR.  We will need to add that but for simplicity of review, I think it will be better to leave that out for now, especially since I've switched the cache to be namespace-local (I'll have to add the Manager provided cache back in to access the `service-ca`)